src/engine/SpatialJoin.cpp

```diff
@@ -0,0 +1,295 @@
+          col += 1;
+        }
+        // add columns from the right id table
+        for (size_t m = 0; m < idRight->numColumns(); m++) {
+          idtable[i * numRowsLeft + k][col] = idRight->at(k, m);
+          col += 1;
+        }
+      }
+    }
+
+    LocalVocab lv = {}; // let's assume, this has no local vocabs
+    std::vector<ColumnIndex> sortedBy = {0};
+    return ResultTable(std::move(idtable), {}, std::move(lv));
+  } else {
+    std::shared_ptr<const ResultTable> res_left_ = childLeft->getResult();
+    std::shared_ptr<const ResultTable> res_right_ = childRight->getResult();
+    const IdTable* res_left = &res_left_->idTable();
+    const IdTable* res_right = &res_right_->idTable();
+    size_t numColumns = getResultWidth();
+    IdTable result{numColumns, _allocator};
+
+    /* this lambda function copies elements from copyFrom
+    * into the table res. It copies them into the row
+    * row_ind_res and column column col_ind_res. It returns the column number
+    * until which elements were copied
+    */
+    auto addColumns = [](IdTable* res,
+                        const IdTable* copyFrom,
+                        size_t row_ind_res,
+                        size_t col_ind_res,
+                        size_t row_ind_copy) {
+      size_t col = 0;
+      while (col < copyFrom->numColumns()) {
+        res->at(row_ind_res, col_ind_res) = 
+                (*copyFrom).at(row_ind_copy, col);
+        col_ind_res += 1;
+        col += 1;
+      }
+      return col_ind_res;
+    };
+
+    // cartesian product with a distance of at most maxDist between the two
+    // objects
+    size_t resrow = 0;
+    auto varColMapLeft = childLeft->getRootOperation()
+          ->getExternallyVisibleVariableColumns();
+    auto varColMapRight = childRight->getRootOperation()
+          ->getExternallyVisibleVariableColumns();
+    ColumnIndex leftJoinCol
+            = varColMapLeft[leftChildVariable.value()].columnIndex_;
+    ColumnIndex rightJoinCol
+            = varColMapRight[rightChildVariable.value()].columnIndex_;
+    for (size_t rowLeft = 0; rowLeft < res_left->size(); rowLeft++) {
+      for (size_t rowRight = 0; rowRight < res_right->size(); rowRight++) {
+        size_t rescol = 0;
+        double dist = computeDist(res_left, res_right, rowLeft, rowRight,
+              leftJoinCol, rightJoinCol);
+        if (dist < maxDist) {
+          result.emplace_back();
+          if (addDistToResult) {
+            result.at(resrow, rescol) = ValueId::makeFromDouble(dist);
+            rescol += 1;
+          }
+          // add columns to result table
+          rescol = addColumns(&result, res_left, resrow, rescol,
+                              rowLeft);
+          rescol = addColumns(&result, res_right, resrow, rescol,
+                              rowRight);
+          resrow += 1;
+        }
+      }
+    }
+    return ResultTable(std::move(result),
+            std::vector<ColumnIndex>{}, LocalVocab{});
```

Comment:
That is wrong, but there is the function that is called for all the other joins, that takes care of the child vocabularies (`ResultTable::localVocabNonEmptyBlaSomething`) which should be used here.